lib/extensions/ar_visible_attribute.rb

```diff
@@ -0,0 +1,23 @@
+module ArVisibleAttribute
+  extend ActiveSupport::Concern
+
+  included do
+    cattr_accessor :hidden_attribute_names
+    self.hidden_attribute_names = []
+  end
+
+  class_methods do
+    # @param [String|Symbol] attribute name of attribute to be hidden from the api and reporting
+    # this attribute is accessible to all ruby methods. But it is not advertised.
+    # we do this when deprecating an attribute or when introducing an internal attribute
+    def hide_attribute(attribute)
+      hidden_attribute_names << attribute.to_s
+    end
+
+    # @return Array[String] attribute names that can be advertised in the api and reporting
+    # Other attributes are accessible, they are just no longer in our public api (or never were)
+    def visible_attribute_names
+      attribute_names - hidden_attribute_names
```

Comment:
I think you also need something like `attribute_names - self.hidden_attribute_names`

Reply:
thnx. I'll keep in mind when running specs if this is necessary